app/src/main/java/com/example/geoquiz/QuizViewModel.kt

```diff
@@ -0,0 +1,32 @@
+package com.example.geoquiz
+
+import androidx.lifecycle.ViewModel
+
+class QuizViewModel : ViewModel() {
+
+    val questionBank = listOf(
+        Question(R.string.question_australia, true),
+        Question(R.string.question_oceans, true),
+        Question(R.string.question_africa, false),
+        Question(R.string.question_mideast, false),
+        Question(R.string.question_americas, true),
+        Question(R.string.question_asia, true)
+    )
+    var currentIndex = 0
+    val currentQuestionAnswer: Boolean
+        get() = questionBank[currentIndex].answer
+    val currentQuestionText: Int
+        get() = questionBank[currentIndex].textResId
+
+    fun moveToNext() {
+        currentIndex = (currentIndex + 1) % questionBank.size
+    }
+
+    fun checkAnswer(userAnswer: Boolean): Int {
+        if (currentQuestionAnswer == userAnswer) {
```

Comment:
Also optional, just a code flavor thing, the return can be lifted out of the if statement. This will then look like 
       `if (currentQuestionAnswer == userAnswer) {
            return R.string.correct
        } else {
            return R.string.incorrect
        }`
        
        Android Studio automatically suggests this change, if you hover mouse over `if` you will get that change suggestion. Clicking on it will automatically do it for you ;)